Sources/UI/RoverViewController.swift

```diff
@@ -51,17 +53,26 @@ open class RoverViewController: UIViewController {
         return cancelButton
     }()
     
-    public init(identifier: ExperienceIdentifier) {
+    public init(identifier: ExperienceIdentifier, campaignId: String? = nil) {
         self.identifier = identifier
+        self.campaignId = campaignId
         super.init(nibName: nil, bundle: nil)
         
         configureView()
         layoutActivityIndicator()
         layoutCancelButton()
     }
     
-    public convenience init(experienceId: String) {
-        self.init(identifier: .experienceID(id: experienceId))
+    public convenience init(experienceId: String, campaignId: String? = nil) {
+        self.init(identifier: .experienceID(id: experienceId), campaignId: campaignId)
+    }
+    
+    public convenience init?(experienceUrl: String, campaignId: String? = nil) {
```

Comment:
Is there a reason the experienceURL is of type String instead of URL? You're unnecessarily making it a failable initializer and the API is actually more cumbersome to work with. In your Example app you receive a variable of type URL, convert it to a string and pass it to this initializer which then converts it back to a URL.

Reply:
👍